### PR TITLE
rValue history of germany

### LINKED ISSUE #95: Feature request: History for R value
Hey, 
I would like to ask you if you could add a history for r value.

### PR DESCRIPTION
**Pull request by PatrickHaussmann**
this would fix #95

There are a few things i am not sure about:

1. is the name `/germany/history/rValue` ok?
2. the first 4 days have a `r-value` of `null`. Should we remove them?
3. `lastUpdate` is set to `now`. Is there a better way get the `lastUpdate` value

### CODE REVIEW

**Comment by Rubber1Duck** [comment]
>     1. is the name `/germany/history/rValue` ok?
> 
>     2. the first 4 days have a `r-value` of `null`. Should we remove them?
> 
>     3. `lastUpdate` is set to `now`. Is there a better way get the `lastUpdate` value
1. think thats okay
2. yes, that makes sense.
3. maybe as in vaccination.ts 
````
const response = await axios.get(
    `https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Daten/Impfquotenmonitoring.xlsx?__blob=publicationFile`,
    {
      responseType: "arraybuffer",
    }
  );
  const data = response.data;
  const lastModified = response.headers["last-modified"];
  const lastUpdate = lastModified != null ? new Date(lastModified) : new Date();
````
that should work .....

File: src/data-requests/r-value.ts [hunk 1]
```diff
@@ -30,16 +35,42 @@ function parseRValue(data: ArrayBuffer): { r: number; date: Date } | null {
...
-    {
-      responseType: "arraybuffer",
-    }
-  );
+  const response = await axios.get(rValueURL, {
+    responseType: "arraybuffer",
+  });
   const data = response.data;
-  const rData = parseRValue(data);
+  if (data.error) {
```
Comment (Rubber1Duck) [inline src/data-requests/r-value.ts:40]: bevor the if block of corse!
  const lastModified = response.headers["last-modified"];
  const lastUpdate = lastModified != null ? new Date(lastModified) : new Date();

File: src/data-requests/r-value.ts [hunk 2]
```diff
@@ -30,16 +35,42 @@ function parseRValue(data: ArrayBuffer): { r: number; date: Date } | null {
...
+
+  var workbook = XLSX.read(data, { type: "buffer", cellDates: true });
+  const sheet = workbook.Sheets[workbook.SheetNames[1]];
+  const json = XLSX.utils.sheet_to_json(sheet);
+  const latestEntry = json[json.length - 1];
+  const rData: RValueEntry = parseRValueRow(latestEntry);
+
   return {
     data: rData.r,
     lastUpdate: rData.date,
```
Comment (Rubber1Duck) [inline src/data-requests/r-value.ts]: lastUpdate: lastUpdate,

**Comment by PatrickHaussmann** [comment]
if we would have:
`const lastUpdate = history[history.length - 1].date;`

then it would be in line with what we return in the function `getRValue`.
Alternatively, we could also get it from the headers in `getRValue`.
what is your take on this?

**Comment by Rubber1Duck** [comment]
both should be the same and it is a good idea to take it from the header! My mind ....

**Comment by Rubber1Duck** [comment]
`/germany/history/rValue/:days` would be very nice!

**Comment by PatrickHaussmann** [comment]
> `/germany/history/rValue/:days` would be very nice!

I am already in the process :)

**Review by Rubber1Duck — CHANGES_REQUESTED** [review]
Dont forget README.md!

**Comment by PatrickHaussmann** [comment]
i am very sorry that it is not clear to me. but what should i change/add in `README.md`?

**Review by Rubber1Duck — APPROVED** [review]
Sorry, there is no changes needed in README.md!
My fault

**Comment by Rubber1Duck** [comment]
God job!
for me that looks all fine!

**Comment by Rubber1Duck** [comment]
is replaced by new PR #494

**Pull request closed without merging**